bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,266 @@
+        )
+        plasmod_solver.execute()
+
+        if plot:
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.pprime(plasmod_solver.x), "pprime", "-"
+            )
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.ffprime(plasmod_solver.x), "ffrime", "-"
+            )
+
+        # generate mesh for the Grad-Shafranov solver
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "plasma_face",
+        }
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=True)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        # initialize the Grad-Shafranov solver
+        p_order = gs_options["p_order"]
+        gs_solver = FemGradShafranovFixedBoundary(mesh, p_order=p_order)
+
+        bluemira_print("Solving fixed boundary Grad-Shafranov...")
+
+        gs_solver.solve(
+            plasmod_solver.pprime,
+            plasmod_solver.ffprime,
+            plasmod_solver.I_p,
+            tol=gs_options["tol"],
+            max_iter=gs_options["max_iter"],
+            relaxation=gs_relaxation,
+            verbose_plot=gs_options["verbose_plot"],
+        )
+
+        # create the finer mesh to calculate the isofluxes
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_fine,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {"lcar": lcar_fine, "physical_group": "plasma_face"}
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        points = mesh.coordinates()
+        x2d_data = np.array([gs_solver.psi_norm_2d(x) for x in points])
+
+        # calculate kappa_95 and delta_95
+        r_geo, kappa_95, delta_95 = calculate_plasma_shape_params(
+            points, x2d_data, [np.sqrt(0.95)]
+        )
+        r_geo, kappa_95, delta_95 = r_geo[0], kappa_95[0], delta_95[0]
+
+        # calculate the iteration error
+        err_delta = abs(delta_95 - delta95_t) / delta95_t
+        err_kappa = abs(kappa_95 - kappa95_t) / kappa95_t
+        iter_err = max(err_delta, err_kappa)
+
+        # calculate the new kappa_u and delta_u
+        kappa_u_0 = builder_plasma.params.get_param("kappa_u")
+        delta_u_0 = builder_plasma.params.get_param("delta_u")
+
+        kappa_u = theta * kappa_u_0 * (kappa95_t / kappa_95) + (1 - theta) * kappa_u_0
+        delta_u = theta * delta_u_0 * (delta95_t / delta_95) + (1 - theta) * delta_u_0
```

Comment:
Hmm.. maybe a Newton-Raphson iteration approach might be more robust? I can't really explain why I think that, but I don't think it would be too hard to do here. With the modification above it would also mean a derivative would make sense.